Add support for Dequantize op via mkldnn for IA backend (codegen + DEX)

nbpatel opened the pull request:
Base branch: #1484.
Dequantize from fp32 --> i8/u8

aprocter commented on src/ngraph/runtime/cpu/op/dequantize.cpp:
```diff
@@ -0,0 +1,39 @@
+    : RequiresTensorViewArgs("Dequantize", {input, min, max})
+    , m_element_type(type)
+{
+    add_output(element::f32, input->get_shape());
```
As with #1484, if we elevate this to core we'll need to check the element types and sizes of `min`/`max`, check the element type of `input` to make sure it is a valid type for a quantized, and make sure that `type` is supported.

Krovatkin replied:
we at least should check that `min` and `max` are the same type and they are both scalars and maybe that the output type is f32/f64 and the inputs are integral types?

jbobba replied:
Also check that min and max are constants.

jbobba replied:
Shouldn't we be using type instead of element::f32?

aprocter commented on test/quantize_cpu.cpp:
```diff
@@ -111,3 +112,43 @@ TEST(quantize_cpu, quantize_to_int8)
+
+template <typename T>
+void DequantizeTest(
+    int input, float min, float max, float expected_output, const element::Type& type)
```
You could use `element::Type::from<T>()` instead of passing `type` here.

Krovatkin commented on src/ngraph/runtime/cpu/builder/dequantize.cpp:
```diff
@@ -0,0 +1,85 @@
+                    auto input_desc = mkldnn_utils::get_input_mkldnn_md(node, 0);
+                    auto result_desc = mkldnn_utils::get_output_mkldnn_md(node, 0);
+
+                    auto min_const_op = std::dynamic_pointer_cast<ngraph::op::Constant>(
```
I've just noticed but this logic that pre-computes values for reorder seems to be similar across `cpu_codegen` and `dex`. Maybe we should refactor it into `mkldnn_emitter::build_dequantization`?

jbobba replied:
Why not static_pointer_cast?

jbobba commented on src/ngraph/runtime/cpu/cpu_emitter.cpp:
```diff
@@ -2740,6 +2741,49 @@ namespace ngraph
+                if (runtime::cpu::mkldnn_utils::use_mkldnn_kernel(node))
+                {
+                    auto& mkldnn_emitter = external_function->get_mkldnn_emitter();
+                    auto input_data_desc = mkldnn_utils::get_input_mkldnn_md(node, 0);
```
need to support this op in cpu_layout

nbpatel commented:
Have addressed all the feedback in #1565. Closing this one since it is based of quantization op branch, which needs some licensing stuff to be resolved. Sorry for the confusion.
